#!/usr/bin/env python
PACKAGE = "bot_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("thresh_high_h", int_t, 0, "Thresholding hue high bound", min=0, max=360)
gen.add("thresh_high_s", int_t, 0, "Thresholding saturation high bound", min=0, max=255)
gen.add("thresh_high_v", int_t, 0, "Thresholding value high bound", min=0, max=255)

gen.add("thresh_low_h", int_t, 0, "Thresholding hue low bound", min=0, max=360)
gen.add("thresh_low_s", int_t, 0, "Thresholding saturation low bound", min=0, max=255)
gen.add("thresh_low_v", int_t, 0, "Thresholding value low bound", min=0, max=255)

gen.add("morph_kernel_size", int_t, 0, "Morphological operation kernel size", min=0, max=11)

gen.add("fullness_high", double_t, 0, "Fullness filtering high bound", min=0.0, max=1.0)
gen.add("fullness_low", double_t, 0, "Fullness filtering low bound", min=0.0, max=1.0)

gen.add("simplify_iterations", int_t, 0, "Simplify contours iteration count", min=0, max=50)

exit(gen.generate(PACKAGE, "bot_vision_node", "Vision"))
